### PR TITLE
feat: add support adding/scaling Num. Simplify when adding LCs.

### PR DESCRIPTION
**Pull request by porcuquine**
This PR adds support for:
+ Adding two Nums.
+ Scaling a Num by a scalar.
+ When adding a variable to LinearCombinations, coalescing with any existing component for that variable.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -214,7 +214,21 @@ impl<E: ScalarEngine> Add<(E::Fr, Variable)> for LinearCombination<E> {
...
+
+        for i in 0..self.0.len() {
+            let (v, mut c) = self.0[i];
+
+            if v.0 == var.0 {
+                c.add_assign(&coeff);
+                self.0[i] = (v, c);
+                found = true;
+            }
+        }
```
Comment (dignifiedquire) [inline src/lib.rs]:
```suggestion
        for (v, mut c) in self.0.iter_mut() {
            if v.0 == var.0 {
                c.add_assign(&coeff);
                found = true;
            }
        }
```

File: src/gadgets/num.rs [hunk 2]
```diff
@@ -407,6 +408,43 @@ impl<E: ScalarEngine> Num<E> {
             lc: self.lc + &bit.lc(one, coeff),
         }
     }
+
+    pub fn add(self, other: &Self) -> Self {
```
Comment (dignifiedquire) [inline src/gadgets/num.rs]: can you add some basic tests that these two methods do what you expect them to do please?

Comment (porcuquine) [reply src/gadgets/num.rs]: I actually put them next to the existing tests.

All right, all right.

Comment (porcuquine) [reply src/gadgets/num.rs]: Done.

**Pull request merged**